sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/IThroughputController.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.throughputControl.controller;
+
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import reactor.core.publisher.Mono;
+
+public interface IThroughputController {
+    /**
+     * Decides whether the throughputController can handle the request.
+     * Different level throughput controller will have its own criteria.
+     *
+     * @param request
+     * @return
+     */
+    boolean canHandleRequest(RxDocumentServiceRequest request);
+
+    /**
+     * Close all the scheduled tasks and any other resources need to release.
+     * @return
+     */
+    Mono<Void> close();
+
+    /**
+     * Initialize process.
+     * Will create and initialize the lower level throughput controller and schedule tasks if needed.
+     * @param <T>
+     * @return
+     */
+    <T> Mono<T> init();
```

Comment:
Can we parameterize the interface like -> `IThroughputController<T>` - so that implementations can also follow the same thing, and you don't have to cast in the implementation. 

    @SuppressWarnings("unchecked")

Reply:
I can not easily change to IThroughputController<T> because:
 <T> Mono<T> processRequest: based on which storeModel, the response type will be different